lib/ViewMetaData/ViewMetaData.js

```diff
@@ -1,52 +1,133 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { MetaSection } from '@folio/stripes-components';
+import {
+  withStripes,
+  stripesShape,
+} from '@folio/stripes-core';
 
 class ViewMetaData extends React.Component {
-  static manifest = Object.freeze({
-    createdBy: {
-      type: 'okapi',
-      records: 'users',
-      path: 'users?query=(id==!{metadata.createdByUserId})',
-    },
-    updatedBy: {
-      type: 'okapi',
-      records: 'users',
-      path: 'users?query=(id==!{metadata.updatedByUserId})',
-    },
-  });
-
   static propTypes = {
     metadata: PropTypes.object,
-    resources: PropTypes.shape({
-      createdBy: PropTypes.shape({
-        records: PropTypes.arrayOf(PropTypes.object),
-      }),
-      updatedBy: PropTypes.shape({
-        records: PropTypes.arrayOf(PropTypes.object),
+    stripes: stripesShape.isRequired,
+    systemUser: PropTypes.oneOfType([
+      PropTypes.shape({
+        id: PropTypes.string,
+        personal: PropTypes.shape({
+          firstName: PropTypes.string,
+          lastName: PropTypes.string,
+          middleName: PropTypes.string,
+        }),
       }),
-    }),
+      PropTypes.string,
+    ]),
   };
 
   static defaultProps = {
     metadata: {},
+    systemUser: 'System',
```

Comment:
Can we pull this from a translation file via `FormattedMessage` instead? (And I guess update `propTypes` to support `node` as well.

Reply:
This property is passed to `MetaSection` component so I took PropTypes from there. And `MetaSection` component will not handle `node` correctly.

Reply:
In that case we could just update `MetaSection`'s proptypes to accept a `node` and tweak it's `renderUser` function, no?

We'll definitely need translations for "System" though. I just think that a `FormattedMessage` is the cleanest way of doing it when we don't really _have_ to get it into a string.

Reply:
Agreed; we need translation here. That should be a trivial change in stripes-components, and then we'll need to get that released and bump the dep in the package.json here as well.

Reply:
Alas, it's not completely trivial because of the type-checking (outside of proptype checking) that `MetaSection`'s `renderUser` function does, but definitely still not a burden. _Is_ a "gotcha" though.

Reply:
Hmmmmm. Seeing the ripple effect of [providing i18n support in MetaSection](https://github.com/folio-org/stripes-components/pull/908), I'm increasingly wary of that approach as it clearly relies on their being a hard-coded identifier that corresponds to the "system" user, thus the need for "system" to be localized, rather than there being a user, just like any other, which one tenant could name "system" and another could name "система".

I feel badly that you are doing everything we are asking here, quickly and elegantly to boot, and I'm still hesitant to merge the code. I just really worry that by coding this up, we are implicitly endorsing an approach to handling the system user problem when in fact how to handle that problem is still undecided. 

I dunno; maybe I'm over reacting. @doytch, any thoughts? As you said above, "magic numbers ... yuck, but you gotta do what you gotta do."

Reply:
(context for "the problem is still undecided": [RMB-320](https://issues.folio.org/browse/RMB-320))

Reply:
@zburke, please make a decision about a merge sooner as we need these changes anyways in `ui-data-import` module. In case you are not agree on these changes we simply create this component in our module. But I think this could be the case not only for `ui-data-import` app but for any app where a similar issue may occur. I do not think that resolving the same problem per component is a good solution.